### PR TITLE
[BUGFIX] Show FDS Asset name in DataDocs

### LINKED ISSUE #10339: [BUG] Asset name is rendered as empty string in Data Docs
**Describe the bug**
When using checkpoint to validate a pandas data asset based on an in-memory data frame, the resulting validation result rendered in the Data Docs website has an empty asset name.

**To Reproduce**
To reproduce the error, you can run this single script in its entirety along with the data file I have attached to this post. Within it, a new file based context will be created with all default values, so there is no need to share the great_expectations.yml file.

```python
import great_expectations as gx
import pandas as pd

# -- Set GX constants for artifact creation
NAME_DATA_SOURCE = "pandas"
NAME_DATA_ASSET = "tutorial_data"
NAME_BATCH_DEF = "pandas_tutorial"
NAME_EXPECTATION_SUITE = "tutorial"
NAME_VALIDATION_DEF = "pandas-validation"
NAME_CHECKPOINT = "pandas"

# -- Load data for configuration
FILE_CONFIGURE = "data/yellow_tripdata_2021-11.csv"
df_configure = pd.read_csv(FILE_CONFIGURE)

# -- 1. Initialize GX for configuration
context = gx.get_context(mode="file")

data_source = context.data_sources.add_pandas(name=NAME_DATA_SOURCE)
data_asset = data_source.add_dataframe_asset(name=NAME_DATA_ASSET)
batch_definition = data_asset.add_batch_definition_whole_dataframe(NAME_BATCH_DEF)

# -- 2. Configure expectation suite to be called over runtime data later
expectation_suite = gx.ExpectationSuite(name=NAME_EXPECTATION_SUITE)
expectation_suite = context.suites.add(expectation_suite)

# -- 2.1. Define table level expectations
columns = list(df_configure.columns)
expectation = gx.expectations.ExpectTableColumnsToMatchSet(column_set=columns)
expectation_suite.add_expectation(expectation)

# -- 2.2. Define column level expectations
expectation = gx.expectations.ExpectColumnValuesToNotBeNull(column="VendorID")
expectation_suite.add_expectation(expectation)

expected_values = [1, 2]
expectation = gx.expectations.ExpectColumnValuesToBeInSet(
    column="VendorID", value_set=expected_values
)
expectation_suite.add_expectation(expectation)

# -- 2.3. Evaluate results on test dataset
batch_parameters = {"dataframe": df_configure}
batch = batch_definition.get_batch(batch_parameters=batch_parameters)
validation_results = batch.validate(expectation_suite)

# -- 3. Bundle suite and batch into validation definition and checkpoint w/ bundled
# --    actions for easy execution later
validation_definition = gx.ValidationDefinition(
    data=batch_definition, suite=expectation_suite, name=NAME_VALIDATION_DEF
)
validation_definition = context.validation_definitions.add(validation_definition)

action_list = [
    gx.checkpoint.UpdateDataDocsAction(
        name="update_all_data_docs",
    ),
]
checkpoint = gx.Checkpoint(
    name=NAME_CHECKPOINT,
    validation_definitions=[validation_definition],
    actions=action_list,
    result_format={
        "result_format": "COMPLETE",
    },
)
context.checkpoints.add(checkpoint)

# -- 4. Run checkpoint to validate if everything works properly
runid = gx.RunIdentifier(run_name="Configuration run")
results = checkpoint.run(batch_parameters=batch_parameters, run_id=runid)
```

**Expected behavior**
I would expected that asset name of the data asset used to invoke the checkpoint over to show up in the asset name section, but instead it is empty.

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 1.0.1
 - Data Source: Pandas
 - Cloud environment: N/A
 
[yellow_tripdata_2021-11.csv](https://github.com/user-attachments/files/16885718/yellow_tripdata_2021-11.csv)

### PR DESCRIPTION
**Pull request by Kilo59**
- Fixes https://github.com/great-expectations/great_expectations/issues/10339
- V1 version of https://github.com/great-expectations/great_expectations/pull/9950

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -99,6 +99,10 @@
     from typing_extensions import TypeAlias
 
     from great_expectations.checkpoint.checkpoint import CheckpointResult
+    from great_expectations.core.datasource_dict import DatasourceDict
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py:102]:
```suggestion
    from great_expectations.datasource.datasource_dict import DatasourceDict
```

File: tests/render/test_page_renderer.py [hunk 2]
```diff
@@ -526,3 +541,41 @@ def test_snapshot_ValidationResultsPageRenderer_render_with_run_info_at_start(
...
+    validator.expect_column_values_to_not_be_null("pickup_datetime")
+    validator.expect_column_values_to_be_between("passenger_count", min_value=1, max_value=6)
+    validator.save_expectation_suite(discard_failed_expectations=False)
+
+    checkpoint = context.add_or_update_checkpoint(
+        name="my_quickstart_checkpoint",
+        validator=validator,
+    )
+
+    checkpoint.run()
```
Comment (Kilo59) [inline tests/render/test_page_renderer.py]: What's the equivalent quickstart for running a checkpoint in v1?

Edit:
Going with this
https://docs.greatexpectations.io/docs/core/introduction/try_gx

File: tests/render/test_page_renderer.py [hunk 3]
```diff
@@ -522,3 +539,64 @@ def test_snapshot_ValidationResultsPageRenderer_render_with_run_info_at_start(
...
+    """
+    import great_expectations as gx
+    from great_expectations.render.renderer.site_builder import DefaultSiteIndexBuilder
+
+    context = gx.get_context(mode="ephemeral")
+
+    add_resource_info_spy = mocker.spy(
+        DefaultSiteIndexBuilder,
+        "add_resource_info_to_index_links_dict",
+    )
```
Comment (cdkini) [inline tests/render/test_page_renderer.py:558]: Can we test this another way? The public API of the default site index builder isn't super well known this feels like more of an implementation detail?

Comment (Kilo59) [reply tests/render/test_page_renderer.py:558]: I agree it's not great but I don't have another way that doesn't involve a lot more effort actually parsing an HTML Doc.
This is the same way I tested the 0.18 version of this fix.

- https://github.com/great-expectations/great_expectations/pull/9950

**Review by cdkini — APPROVED** [review]

**Pull request merged**